### ISSUE TITLE
FNX-12818 ⁃ As a user, I find the tab experience to be horrible and frustrating

### ISSUE THREAD

**Issue by pwd-github** [mozilla-mobile/fenix#11833]
## User Story
* As a user, I want to be able to manage tabs in a simple and straightforward manner so I can do keep my sites as the focus of my browsing experience. (keep it problem-centric)

## Dependencies
* List dependencies on other issues/teams etc.
* I cannot open a new tab from an existing tab. I'm forced to open the tab tray first. This seems logical, but isn't. 
* There is two ways to access private browsing. The toggle on the main screen and then via the tab tray 
* Top sites aren't available via the new tab tray. 
* Collections aren't available via the new tab tray. 
* The tab tray doesn't indicate whether you have any tabs in the other private/normal browsing list. 
* Private browsing UI doesn't follow the purple colour scheme
* Opening the address bar and then pressing back closes the app 

### Acceptance Criteria
* I can do …  (e.g. add a bookmark)
* Add new tab to overflow menu
* I'm unsure of how to fix this. Maybe just remove the toggle and rely on the overflow menu completely. 
* This will be less frustrating when being able to make new tabs becomes a global experience of the app (i.e. New Tab is added to the overflow menu) 
* As above
* When switching to the private tabs tray, colours should invert. 
* It should take you back to whatever page you were at prior.

**Comment by AmyYLee** [comment]
@topotropic for UX feedback

**Comment by Pajn** [comment]
I agree about the new tabs UI and much preffer the older one. This change is unfortunately pushing me back to Chrome.

**Comment by phildenhoff** [comment]
I have to agree. The new pop-up tab design is a much worse experience than what existed before, and it's super unclear when you open a new tab what's going on.

**Comment by julKali** [comment]
Yes, as soon as I downloaded 5.2.0, I went to GitHub to open an issue just like this one. I have multiple problems with this layout:
1. the popup tray bar doesnt fit the style. It is, in my opinion, too light-weight, too flat to fit in with the rest of the UI elements
2. to get to the main screen with collections and favorite apps, you have to click on the add new tab button ... what?
3. the tabs now take much more screen real estate, so it takes longer to check which tabs are open
4. the new tab buttons in normal vs private tabview are extremely inconsistent. not only is the word "private" in the private tab view unnecessary, the two buttons are also not of the same height.
5. i dont think that normal and private browsing should be listed in the sort of "horizontal menutab list" as it is now. with normal and private browsing, there exists a duality, which cannot be changed. there is only this or that. the menutab design as it is now, would suggest that there may be more than the two browsing modes. i prefer the previous version, because there was a toggle button that would enter an "alternate" UI view that was very much the same as the normal view but with different colors. the toggle button made clear that it is this or the other, no third option allowed.
5. such a design with having normal and private next to each other in this menutab view is also highly frustrating as you cannot swipe between them. this is not possible because the swipe to delete gesture on individual tabs is already in place, so adding a swipe to left gesture on the tab list would make gestures ambiguous. this leads to having to press the "private" menutab at the top which is, on mobile devices, particularly painful, as such a layout is virtually almost done with swiping. On top of that, because the menutab is on the left side of the screen, it is further away from the thumb than the private browsing toggle button in the old version, so you have to stretch across the screen even more.

I have the feeling, whoever came up with this design doesn't use fenix, at least not as their main browser. Usually, change is good. But sometimes, things are already good enough. There are a number of issues that are actually annoying in daily app use and could be fixed with rather little effort. The tab view, however, was refreshingly simple, well-arranged and I valued fenix for it. Please don't destroy what is good, just to change something.

**Comment by pwd-github** [comment]
Let me just add, I like the bottom sheet in concept. Just feel it needs some tweaking. Though, as with the above comments, the simplicity and intuitiveness of the previous tab management experience is one that should be considered when moving forward.

**Comment by JoGarWeb** [comment]
I agree new tabs UI SUCKS!! 🤮🤮🤮🤮
It's very annoying 😠🤬

**Comment by vrubleg** [comment]
I loved the simplicity of the previous UI. The new version looks ugly and confusing.

**Comment by yoasif** [comment]
I think the best way to respond to the feedback here is to focus the address bar on new tab creation, and to include the home page content on that screen.

Right now, the biggest regression from the pre-tab tray version of Fenix is that that version focused the address bar on new tab. This matches desktop, FWIW, but is unlike other Android browsers. 

I think that as a matter of consistency with mobile browsers on Android, it initially made sense to not focus the address bar on new tab. However, upon review (and feedback here) it really does make the Fenix UI feel a lot faster/productive (especially if the use case the user is trying to accomplish is to open a new tab to a new URL or search).

Ideally the flow would look like:

1. Tap tab switcher icon (or open tab switcher menu)
2. Tap new tab FAB or new tab context menu
3. Fenix opens Fenix Home with address bar focused
4. Address bar page dispenses with "this time search with" content and instead includes items from home
5. User can touch the content area to access Home directly (move focus from address bar) OR they can tap back to close the keyboard

Samsung Internet, Edge and Opera all show some piece of the home page content when their address bars are focused - the only difference here would be that Fenix would focus the address bar automatically.

![signal-2020-06-24-160705_003](https://user-images.githubusercontent.com/152493/85622473-dec18300-b634-11ea-9968-f4371de8aead.png)
![signal-2020-06-24-160705_002](https://user-images.githubusercontent.com/152493/85622475-df5a1980-b634-11ea-9e37-50971912fe3d.png)
![signal-2020-06-24-160705_001](https://user-images.githubusercontent.com/152493/85622477-df5a1980-b634-11ea-869d-1911b339c834.png)
 
I think that the tab tray continues to make sense, because the home page isn't really a tab switcher (or it ought not to be, imo) - the tab tray is purposely designed to just be a tab switcher, but because the prior iteration of the tab switcher (home page!) had a + icon where the tab switcher icon was, it felt a lot faster to open a new tab. 

My suggestion here allows users to go back to the two tap method of opening a new tab and focusing the address bar on open (like desktop and pre-tab tray Fenix).

The larger issue with the old design was that opening a new tab never actually opened a new tab unless you navigated to a new page in the address bar "interstitial", which the new design helps along.

**Comment by pwd-github** [comment]
Let me just add, I like the bottom sheet in concept. Just feel it needs some tweaking. Though, as with the above comments, the simplicity and intuitiveness of the previous tab management experience is one that should be considered when moving forward.

I will add that currently it's too many taps/too confusing to open a new tab. I honestly believe that adding "Open New Tab" to the context menu will mitigate most of the issues. Having collections and homescreen shortcuts on the public and private new tab page will help loads too. Perhaps even adding contextual recent tabs would help also.

**Comment by vrubleg** [comment]
A story. Years ago all browsers had separate address bar and search bar. Now they are merged into one universal entity, because Google was brave enough advertise it as their main feature. Funny fact: Firefox (Phoenix) v0.1 had the same feature in 2002, years before Chrome. But the developers weren't brave enough, and the separate search bar was added before v1.0 was released. Today, everybody think that it was introduced in Chrome.

Same thing here. It was really a great idea to put the tabs list to the main page and remove a useless entity. If Mozilla is not brave enough to advertise this new cool thing as their main feature, somebody else will do it one day, but it will be too late for Mozilla.

**Comment by cadeyrn** [comment]
>  It was really a great idea to put the tabs list to the main page. Probably, the best idea for a mobile browser in a decade.

I would definitely not go that far. It was a nice idea but not "the best idea for a mobile browser in a decade". FWIW I really like the new tabs tray (module some things that should be improved, of course).

> If Mozilla is not brave enough to advertise this new cool thing as their main feature, somebody else will do it one day, but it will be too late for Mozilla.

Even if this were true it wouldn't be "too late" only because someone else already did this. I don't know why you think it's important to be the first.

**Comment by vrubleg** [comment]
Chrome was released in 2008, 6 years after the Mozilla Phoenix v0.1. The idea became widely adopted only after it was advertised by Google. Most people need explanation and some time to get used to some new great ideas, to understand why it is really so great. Eventually, everybody understood how convenient to use the address bar for searching. These days there are no major browsers which use separate search bar by default. But people were blind from 2002 to 2008. Just because nobody at Mozilla even tried to promote this idea, to explain why it is so cool. Google did it in 2008 and later. Today, most of people consider Firefox as a Chrome rip-off. Just because Google noticed some really good ideas, put them into the front, and advertised them as much as they could. People loved these ideas, and other browsers had to follow.

**Comment by julKali** [comment]
@cadeyrn can you elaborate on why you didn't like the old view? In my opinion, the main page now gets obsolete because collections and favorite sites aren't enough for one entire view. f.e. when i open firefox, i see a lot of unused space because i dont have any collections.
In my view, the new tab tray just adds a lot of complexity because it introduces new UI without clear benefit (to me). This is why I am genuinely curious about why this change was made in the first place. What is better now in terms of UX?

**Comment by vrubleg** [comment]
@julKali I also don't use collections at all. I just keep all the frequently used pages always open in tabs, so I didn't even use bookmarks, because the tabs were so convenient to be used for everything.

This approach won't ever work on desktop, but it seems like a genius idea for a mobile browser. I loved it almost immediately after I tried the Firefox Preview, and I was really proud of Mozilla that they tried to innovate something really new.

**Comment by julKali** [comment]
@vrubleg yes, I used it exactly the same way! No thinking about collections, no having to open any history, library, etc, just a bunch of links that I want to revisit sometime later, neatly aligned as soon as I open the app. You can scroll through the list, know immediately what every tab was about and when you need a new one, you can just click on the search bar right next to your thumb - easy! And from any open web page it's just one click to the main screen.
I, too, was very much delighted about this novel approach and it would indeed be a shame to move away from all of that :(

**Comment by worg** [comment]
Having the list of tabs missing from the start page was (still is) confusing (and frustrating), tab management was really cool that way, and I really hope you reconsider bringing it back (along with keeping the drawer with improvements)

**Comment by kannan936** [comment]
New tab management is worst

**Comment by kannan936** [comment]
New tab management is worst

**Comment by vrubleg** [comment]
An addition. Tabs are the most important part of the browser after the address bar, and they must be really easily accessible, as much as possible. Collections and top sites could be removed from the browser completely. Just allow to pin some opened tabs to the top of the list of opened tabs, and allow to reorder opened tabs and making group of them (by moving one tab on another, as you do folders on Android homescreen). The group of tabs could be collapsed and uncollapsed by tapping on the header of the group. So, if anybody finds the idea of collections useful, they'll use this simple UI to create such collections. This way, you keep the most important things on the first places, and won't loose any other functionality you have on the main screen now. Just because tabs, top sites, and collections are merged into one entity on the main page of the browser which is always available. And you don't even need to think if you want to keep some tabs open: just keep it open if you think that you'll return back. It works as bookmarks also! The browser don't need to keep all the opened tabs in memory. Firefox already unloads the unused tabs, so no need to change anything here.

**Comment by SS1113** [comment]
I want to say that I like the new Tabs Tray a lot than the old one. It makes the homescreen less cluttered. And I can access other features of the browser easily like **Top Sites** and **Collections** and other features that'll coming to the homescreen eventually.

The **Top Sites** and the **Collections** have become really important features for me! And I don't want to have my homescreen clutter with the tabs let's say more than 30 or 50, it'll make it harder to reach the top or bottom of the homescreen and I'll have to scroll and scroll and scroll...!  . Of course there are some issues with the new tabs tray bit they can be solved (I haven't able to pinpoint the issue yet, but for some reason it feels a little unintuitive when interacting with it when opening the first new tab). I think the new Tabs Tray behaviour should not be removed but instead improve upon the usefulness/intuitiveness it had in the old tabs layout.

Tab management is a one of the central features of a browser as our @vrubleg say and I agree with them (you can see that I have filed some [issues](https://github.com/mozilla-mobile/fenix/issues/11376) regarding this as well), they need to really easy to use since everyone will use them a lot!

**Comment by ListeningGarden** [comment]
The previous behavior of the tabs existing within the main screen was one of the smoothest parts of the experience with Fenix - when the browser was booted up, it allowed me to choose what page to load first instead of assuming the last page I had open was what I wanted, and there were less button presses involved in order to perform certain actions like entering the URL for a new tab.

Right now, there is a bit of a regression with the new tab tray in that I now require 3 taps instead of 2 in order to begin the process of a new tab, as well as a disconnect between the iconography and the expected behavior compared to what the button actually does.

Old method: Click on tabs button in bottom corner of the UI, tap URL Bar. I can now begin to enter my URL.
New method: Click on tabs button in bottom corner of the UI, tap the new tab button, confusingly be taken to the collections tray/main screen instead, require an additional tap to access URL bar. I can now begin to enter my URL.

Collections are certainly important to Fenix's identity, so the user will still need a way to access those, right? But, when I click on the "+" button in the tab tray, I expect to be taken directly to a URL bar so that I can quickly and more fluidly access the site I wish to visit. In order to provide a fair experience to users who both wish to enter a URL directly, as well as those who wish to access their collections & topsites, two buttons should be available on the tab tray if the tab tray must be kept. One button, the "+" button, would take the user directly to the URL prompt and bypass the main screen entirely. The second - the iconography of which would need to be determined to provide a natural metaphor to match the behavior - would take the user to the main screen, allowing them to access their collections and topsites. This way, we trim the redundancy users experience in wanting to visit a new site or require manually entering or pasting a URL, without hiding or obscuring the way to access collections & topsites.

Ideally, things would be designed to require as few steps as possible for the user to initialize the action they desire at any given moment. The current tab tray could still work quite well, but it requires a bit more designing to feel less sharp (the old tabs were round and kinda cute, without being unprofessional imo for those who prefer a more "serious" look for their browser) and fit the overall aesthetics of Fenix better while also ensuring UX does not take a step backwards when trying to accommodate for potentially complex use cases and feature access. 

That said, overall I feel as though the old way got it right and provided the smoothest experience overall for how I personally use the browser, so I wouldn't mind too much if tabs and collections were merged again and the main screen behaved closer to how it once did without the additional tab tray. My two cents on a complicated, behavioral issue.

**Comment by SS1113** [comment]
>Right now, there is a bit of a regression with the new tab tray in that I now require 3 taps instead of 2 in order to begin the process of a new tab

I agree with you here. Something in the new Tabs Tray feels really unintuitive to me. And this could be it!

**Comment by vrubleg** [comment]
@SS1113 did the users of the Firefox Preview complain about the previous tab management system on the main screen much? Or they just were happily using it? If all the browsers use the same UX approach, what is the point of existence of Firefox? I would just use default browser in this case. And I was using the default browser before Firefox Preview.

Firefox Preview doesn't have anything else useful on the main page. Collections and top sites can be merged into the tab list, I already mentioned it. There is nothing else. If Mozilla wants to add some garbage like "latest news" and advertisements, they could just put it to a separate page which is accessible from the main page (it could be just a small button next to the Firefox logo).

**Comment by julKali** [comment]
@SS1113 you talking about your situation with having to scroll through many tabs in order to reach your collections made me realise what the intentions behind this change were. However, as collections are undoubtetly less used by the broad userbase than tabs, I think it would make more sense to instead move collections somewhere else where they can be accessed more easily. One could think of a swipe-to-left gesture that would open a sidebar-like panel that contains a user's saved collections. I strongly believe that moving the tab list instead was the wrong approach.

As has been mentioned, the new tab list is also annoyingly uninnovative. It frustrates me to see Fenix surrender to the Android layout guidelines instead of nudging users into a more creative but, in my case, convenient way of using tabs. I noticed that seeing all the tabs at startup, I closed much more unneeded ones than with previous browsers. This is because tabs were always hidden and you would maybe see the top few tabs once in a while. While I easily reached 100 tabs in the old Fennec, I now have roughly 12 open at any time.

**Comment by SS1113** [comment]
@vrubleg 

>did the users of the Firefox Preview complain about the previous tab management system on the main screen much? 

I didn't like it that much but I feel what you are saying; after getting accustomed to using the old tabs layout and finding it really useful and now  you have to use the new layout suddenly! I get you!

>If all the browsers use the same UX approach, what is the point of existence of Firefox? 

To provide you and me and others with the control and the choice over how we browse the Web!

>Collections and top sites can be merged into the tab list

Please don't do that! I like it how it is right now!

But I still get what you are saying about the new tabs tray. And when and if Mozilla reverts to the old tabs tray layout, I want to make it clear that _please_ make it optional so that I can go back to using the new Tabs Tray layout! I have already become used to the new Tabs Tray now and I like it very much!

**Comment by pwd-github** [comment]
If I'm honest, I'm flabbergasted that such a poor feature as collections continues to get the support it does. It would be one thing if they were working properly. i.e. Collections should act like dynamic tab groups, not bookmark folders. But they're actually just fundamentally rubbish. They're glorified bookmarks. They should not be front and centre of any design or marketing decisions. 

But I digress. The new bottom sheet tab draw is in line with modern Android design. Addressing the suggestions in the first post fixes most usability issues. I say this as someone that's been using them on nightly since they were hidden behind the five tap experiment thing. I realise that it's easy to assume that this was just a knee-jerk reaction to the stable release, so I wanted to clear that up. 

Regarding the landing page of the app or the new tab page, wherever you'd like to call it. I have considered some of the feedback here and would like to raise a suggestion. What about that page featuring recent and synched tabs sections?

**Comment by pwd-github** [comment]
Bookmarks remain a highly useful browser feature. Rebranding them as collections though... 😒

**Comment by jensrutschmann** [comment]
I'd like to add that the new tab (+) button is very hard to reach when the url bar is placed at the top rather than bottom. This combined with the extra tap for creating a new tab also makes using the new version quite annoying.

Previously I just had to tap the tab icon and the the url bar the start typing. Done.

Now I need to tap the tab icon, move my thumb all the way to the bottom (and hold the precious huge modern phone with the other hand so it doesn't fall to the floor), tap the plus button, move the thumb all the way back to the top.

Working with tabs is very important to me and I just switched from chrome to fenix because of the smooth experience even though I lost pull-to-refresh and swiping on the url bar for switching tabs.

Sorry if this sounds like a rant but I was just so happy about the UX of fenix. It was the perfect android browser (just lacking those swipe gestures). And then came tabs tray ... 😞

**Comment by HollMax** [comment]
Feels like someone was unhappy with the old tabs and decided to redesign them, so now he's happy with his tabs. And that's fine, the old tabs weren't perfect. But here's the thing, whoever did the redesign completely ignored what made the old tabs great for many of its users, so instead of  improved tabs for everybody, we ended up with maybe better tabs for someone, but surely worse tabs for me and everyone else in this thread.
I want the old tabs back, and to have a discussion about what the goal of this was, so we can find a solution, that doesn't screw up the user experience.

**Comment by cadeyrn** [comment]
> Feels like someone was unhappy with the old tabs and decided to redesign them, so now he's happy with his tabs. […] whoever did the redesign completely ignored

Such comments are not helpful at all…

> surely worse tabs for me and everyone else in this thread

It's totally okay to say that you don't like it but please don't talk for me - I am part of "everyone else in this thread" and since I already said that I am not against the tabs tray this statement is simply wrong. Sure, there are things that can and should be improved but it doesn't mean that the old implementation has to be restored.

**Comment by ekager** [comment]
Thanks for your passion and feedback! If you want specific issues about tabs tray addressed please open individual issues for each bug or feature suggestion. Meta issues like this will likely never be looked at.

**Comment by ekager** [comment]
* I cannot open a new tab from an existing tab. I'm forced to open the tab tray first. This seems logical, but isn't.
 
You can long press the tab icon in the toolbar and open a new tab now (#11569 )

* Top sites aren't available via the new tab tray.
* Collections aren't available via the new tab tray.

Opened https://github.com/mozilla-mobile/fenix/issues/11958

* The tab tray doesn't indicate whether you have any tabs in the other private/normal browsing list.
 Opened https://github.com/mozilla-mobile/fenix/issues/11959

* Private browsing UI doesn't follow the purple colour scheme
Opened https://github.com/mozilla-mobile/fenix/issues/11960

* Opening the address bar and then pressing back closes the app
 Is this #11866  ?

* Add new tab to overflow menu
Opened #11961
